tests/tests.rs

```diff
@@ -106,3 +106,13 @@ fn bind_device() {
         .send()
         .unwrap();
 }
+
+#[test]
+fn duration() {
+    // Ensure that the duration returned is less than the time elapsed 
```

Comment:
Unfortunately, we can't send dgram icmp package on GitHub Actions, so please add `skip_if_no_capability!();` here to skip it.